pandas/core/indexes/base.py

```diff
@@ -4866,6 +4867,20 @@ def slice_locs(self, start=None, end=None, step=None, kind=None):
             # If it's a reverse slice, temporarily swap bounds.
             start, end = end, start
 
+        # GH 16785: If start and end happen to be date strings with UTC offsets
+        # attempt to parse and check that the offsets are the same
+        if (isinstance(start, compat.string_types) and
+                isinstance(end, compat.string_types)):
+            try:
+                ts_start = Timestamp(start)
+                ts_end = Timestamp(end)
+            except ValueError:
+                pass
+            else:
+                if not tz_compare(ts_start.tz, ts_end.tz):
```

Comment:
shouldn't this happen regardless if these are strings or not?

e.g. df[pd.Timestamp(...):'string']

Reply:
Hmm didn't know we supported `df[pd.Timestamp(...):'date_string']`. If that's the case then definitely.